exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -568,182 +883,362 @@ func (e *Exchange) CancelBatchOrders(ctx context.Context, o []order.Cancel) (*or
 		default:
 			return nil, order.ErrOrderIDNotSet
 		}
-	}
-	cancelledOrders, err := e.CancelMultipleOrdersByIDs(ctx, orderIDs, clientOrderIDs)
-	if err != nil {
-		return nil, err
+		if assetType == asset.Futures {
+			if o[i].Pair.IsEmpty() {
+				return nil, currency.ErrSymbolStringEmpty
+			} else if o[0].Pair != o[i].Pair { //nolint:gosec // index 0 safe because loop requires len(o) > 0
+				return nil, currency.ErrPairNotFound
```

Comment:
🚧 Change request:
This error is not correct.
But also: Once we've partitioned the batch by type, we can partition it again by Pair and make separate requests to cancel.
I really think we should, or we need to write a tonne of conditions into the method documentaton.

Reply:
This seems like it was actioned. Resolving.